man/sd_device_enumerator_add_all_parents.xml

```diff
@@ -0,0 +1,204 @@
+    <parameter>subsystem</parameter> are enumerated. The <parameter>match</parameter> parameter is a
+    boolean flag (0 = exclude, 1 = include).</para>
+
+    <para>The <function>sd_device_enumerator_add_match_sysattr()</function> function adds a filter on
+    the sysfs attribute <parameter>sysattr</parameter> matching <parameter>value</parameter>. The
+    <parameter>match</parameter> parameter is boolean (0 = exclude, 1 = include).</para>
+
+    <para>The <function>sd_device_enumerator_add_match_sysname()</function> function adds a filter so
+    that only devices whose sysname equals <parameter>sysname</parameter> are enumerated.</para>
+
+    <para>The <function>sd_device_enumerator_add_match_tag()</function> function adds a filter so that
+    only devices tagged with <parameter>tag</parameter> are enumerated.</para>
+
+    <para>The <function>sd_device_enumerator_add_nomatch_sysname()</function> function adds a filter so
+    that devices whose sysname equals <parameter>sysname</parameter> are excluded from the
+    enumeration. This is useful for excluding specific devices from the enumeration process.</para>
+
+    <para>The <function>sd_device_enumerator_allow_uninitialized()</function> function allows devices
+    that have not yet been initialized by udev to be included in the enumeration.</para>
+  </refsect1>
+
+  <refsect1>
+    <title>Return Value</title>
+
+    <para> All functions return 0 on success or a negative errno-style error code on failure.</para>
+    <refsect2>
+    <title>Errors</title>
+
+    <para>Returned errors may indicate the following problems:</para>
+
+    <variablelist>
+      <varlistentry>
+        <term><constant>-ENOMEM</constant></term>
+
+        <listitem><para>Memory allocation failed.</para></listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><constant>-EINVAL</constant></term>
+
+        <listitem><para>The argument is invalid.</para></listitem>
+      </varlistentry>
+    </variablelist>
+  </refsect2>
+  </refsect1>
+
+  <refsect1>
+  <title>Examples</title>
+
+  <example>
+    <title>Detect Removable USB Devices (Using Match and Exclude)</title>
+    <programlisting><xi:include href="sd_device_enumerator_add_all_parents-example.c" parse="text"/></programlisting>
+  </example>
+  </refsect1>
+
+  <refsect1>
+    <title>History</title>
+
+    <para><function>sd_device_enumerator_add_all_parents()</function>,
+    <function>sd_device_enumerator_add_match_parent()</function>,
+    <function>sd_device_enumerator_add_match_property()</function>,
+    <function>sd_device_enumerator_add_match_property_required()</function>,
+    <function>sd_device_enumerator_add_match_subsystem()</function>,
+    <function>sd_device_enumerator_add_match_sysattr()</function>,
+    <function>sd_device_enumerator_add_match_sysname()</function>,
+    <function>sd_device_enumerator_add_match_tag()</function>,
+    <function>sd_device_enumerator_add_nomatch_sysname()</function>, and
```

Comment:
sd_device_enumerator_add_nomatch_sysname() is since v251

You can easily find that in src/libsystemd/libsystemd.sym